sdk/core/core-rest-pipeline/src/util/stream.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Readable } from "stream";
+import { BlobLike } from "../interfaces";
+import {
+  isInMemoryBlob,
+  isNodeReadableStream,
+  isStreamableBlob,
+  isWebReadableStream,
+} from "./typeGuards";
+
+function nodeStreamFromWebStream(webStream: ReadableStream): NodeJS.ReadableStream {
+  return Readable.from(
+    (async function* () {
+      const reader = webStream.getReader();
```

Comment:
In that case if we can't use the fromWeb, then we should at least get a lock on the reader.

See:

```js
async function* getData(stream) {
  const reader = stream.getReader()
  try {
    while (true) {
      const {done, value} = await reader.read()
      if (done) return
      yield value
    }
  }
  finally {
    reader.releaseLock()
  }
}
```